### PR TITLE
[16.0][MIG] account_sequence_option: Migration to 16.0 

### PR DESCRIPTION
**Pull request by matteoopenf**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please rename PR title as I have told you in the other one.

**Comment by matteoopenf** [comment]
> Please rename PR title as I have told you in the other one.

done

**Comment by matteoopenf** [comment]
@pedrobaeza  or @kittiu  have any idea how can fix the test or the compute_name() because we do a lot of try but we don't find a solution

**Comment by pedrobaeza** [comment]
Sorry, I haven't dug on such module, and in fact, I prefer core solution instead of this.

**Comment by patrickt-oforce** [comment]
> Sorry, I haven't dug on such module, and in fact, I prefer core solution instead of this.

We prefer core solution, too; but some of our customer has a strange sequence format, with progressive number in front and year as suffix; the new heuristic search of Odoo, confuse the year as progressive number and increment that at every new move confirmed. The problem arise because the customer cannot change the sequencing due to italian fiscal rule

**Comment by pedrobaeza** [comment]
AFAIK, there's a sequence regex you can use for that.

**Comment by patrickt-oforce** [comment]
> AFAIK, there's a sequence regex you can use for that.

Our use case is something similar to <prog_number>/year; what si the regex taht you talk about, can you link the Odoo source?

**Comment by patel1008b** [comment]
@matteoopenf @patrickt-oforce thanks for migration, any chance to merge this PR ?

**Comment by matteoopenf** [comment]
For us the migration is ready and we use the module in production.We need only the review  to merge Matteo ***@***.*** Il giorno 23 nov 2023, alle ore 07:39, bhaveshselarka ***@***.***> ha scritto:﻿
@matteoopenf @patrickt-oforce thanks for migration, any chance to merge this PR ?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

Openforce Srls Unipersonale | www.openforce.it | 02 21119472 | Via Tiziano,32 20145 Milano

**Comment by matteoopenf** [comment]
> @matteoopenf @patrickt-oforce thanks for migration, any chance to merge this PR ?

You can check the code and Run bot and after approve if you agree

File: account_sequence_option/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    )
+
+    @api.depends("posted_before", "state", "journal_id", "date")
+    def _compute_name(self):
+        options = self.env["ir.sequence.option.line"].get_model_options(self._name)
+        move_to_update_name = self.filtered(
+            lambda mv: mv.name not in (False, "/") and mv.state == "posted"
+        )
+        if not options or not move_to_update_name:
+            return super()._compute_name
```
Comment (Saran440) [inline account_sequence_option/models/account_move.py]:
```suggestion
            return super()._compute_name()
```

**Comment by Saran440** [comment]
@matteoopenf Functional Test, this PR is not working. after I add () in my review, it still not working.

I tested by use code v15 it work https://github.com/OCA/account-financial-tools/pull/1635

The problem seems to be related to a conflict between the `account_move_name_sequence` and `account_sequence_option` modules. I looking for solution for this.

**Comment by matteoopenf** [comment]
> @matteoopenf Functional Test, this PR is not working. after I add () in my review, it still not working.
> 
> I tested by use code v15 it work #1635
> 
> The problem seems to be related to a conflict between the `account_move_name_sequence` and `account_sequence_option` modules. I looking for solution for this.

thank you

File: account_sequence_option/models/account_move.py (hunk 1 above)
Comment (matteoopenf) [reply account_sequence_option/models/account_move.py]: thank you

**Comment by Saran440** [comment]
It still not working.

Step to error:
1. install module account and account_sequence_option
2. active sequence option with demo data
3. go to Invoicing > Customers > Invoices
4. create new invoice > Confirm
5. name still **INV/2024/XXXX**. it should be **CINV/XXXX**
Selection_010
Selection_011

I created new PR https://github.com/OCA/account-financial-tools/pull/1810
@matteoopenf  what do you think?

**Comment by matteoopenf** [comment]
> It still not working.
> 
> Step to error:
> 
> 1. install module account and account_sequence_option
> 2. active sequence option with demo data
> 3. go to Invoicing > Customers > Invoices
> 4. create new invoice > Confirm
> 5. name still **INV/2024/XXXX**. it should be **CINV/XXXX**
>    Selection_010
>    Selection_011
> 
> I created new PR #1810 @matteoopenf what do you think?

sorry for my delay Now I check.

**Pull request closed without merging**